src/systems/filecoin_markets/storage_market/storage_market_actor.id

```diff
@@ -17,119 +16,96 @@ type StorageMarketActorState struct {
-    ActivateDeals(rt Runtime, dealIDs [deal.DealID]) [deal.OnChainDeal]
-
-    // call by StorageMinerActor
-    // slash deal collateral depending on the faults (only TerminatedFault)
-    ProcessDealSlash(
-        rt         Runtime
-        dealIDs    [deal.DealID]
-        faultType  sector.StorageFaultType
+    // Deposits the specified amount into the balance held in escrow.
+    // Note: the amount is included implicitly in the message.
+    AddBalance(rt Runtime)
+
+    // Attempt to withdraw the specified amount from the balance held in escrow.
+    // If less than the specified amount is available, yields the entire available balance.
+    WithdrawBalance(rt Runtime, amount actor.TokenAmount)
+
+    // Publish a new set of storage deals (not yet included in a sector).
+    PublishStorageDeals(rt Runtime, deals [deal.StorageDeal])
+
+    // Verify that a given set of storage deals is valid for a sector currently being PreCommitted.
+    VerifyDealsOnSectorPreCommit(
+        rt          Runtime
+        dealIDs     deal.DealIDs
+        sectorInfo  sector.SectorPreCommitInfo
     )
 
-    // batch process storage deal payment
-    // Note: if a sector enters into a Failing state but later recovers, client still pays miner as usual
-    // client only stops paying when a sector gets cleared (expired or faulted)
-    // Note: Clients stop paying when miners enter into Failing state
-    // payment happens at next _onSuccessfulPoSt when miner recovers
-    // If a miner did not recover, payment will stop at the first fault
-    ProcessDealPayment(
-        rt               Runtime
-        dealIDs          [deal.DealID]
-        newPaymentEpoch  block.ChainEpoch
+    // Verify that a given set of storage deals is valid for a sector currently being ProveCommitted,
+    // and update the market's internal state accordingly.
+    UpdateDealsOnSectorProveCommit(
+        rt          Runtime
+        dealIDs     deal.DealIDs
+        sectorInfo  sector.SectorProveCommitInfo
     )
 
-    // batch process storage deal expiration
-    ProcessDealExpiration(rt Runtime, dealIDs [deal.DealID])
-
-    GetPieceInfosForDealIDs(rt Runtime, dealIDs [deal.DealID]) [sector.PieceInfo]
-
-    ClearInactiveDealIDs(rt Runtime, dealIDs [deal.DealID])
+    // Get the weight for a given set of storage deals.
+    // The weight is defined as the sum, over all deals in the set, of the product of its size
+    // with its duration. This quantity may be an input into the functions specifying block reward,
+    // sector power, collateral, and/or other parameters.
+    GetWeightForDealSet(rt Runtime, dealIDs deal.DealIDs) int
+
+    // Terminate a set of deals in response to their containing sector being terminated.
+    // Slash provider collateral, refund client collateral, and refund partial unpaid escrow
+    // amount to client.
+    TerminateDealsOnSlashProviderSector(rt Runtime, dealIDs deal.DealIDs)
+
+    // TODO: previously read:
```

Comment:
I see. In that case, how about if StoragePowerActor keeps track of the epoch in which the slashed sector entered the failing state, and forwards that along with the slash call? Then we can "back-date" the slash appropriately. (Deferring to a future PR for velocity?)

Reply:
sg, the closest we have here is lastPoStSuccessEpoch in StorageMinerActor. Once a sector exceeds MaxFaultCount in StorageMinerActor, deal collateral slashing will be triggered.

Reply:
See comment below that one of _rtUpdatePendingDealStates needs to take in or query lastPoStSuccessEpoch

Reply:
I don't think lastPoStSuccessEpoch is the relevant quantity though, is it? We shouldn't truncate deal payment just because the containing sector happened not to be challenged.

Reply:
In any event, I think it has to be some parameter associated to when the sector changes state in StoragePowerActor; probably we can defer the decision and leave a placeholder for now.

Reply:
we can discuss this, but miners should only be paid for the epochs that they have proven?